src/crimson/osd/pg.cc

```diff
@@ -1519,6 +1543,7 @@ void PG::on_change(ceph::os::Transaction &t) {
   recovery_backend->on_peering_interval_change(t);
   backend->on_actingset_changed(is_primary());
   wait_for_active_blocker.unblock();
+  std::ignore = snap_trim_event_gate.close();
```

Comment:
I'm not sure I understand the purpose of this commit -- we aren't blocking on the gate so we don't prevent the snap trim event dispatch from continuing after the on_change call.  Moreover, I don't actually want to block on all outstanding events completing -- we generally choose instead to check for interval changes in such events using interruptible_future.